src/librustdoc/html/markdown.rs

```diff
@@ -1009,9 +1010,11 @@ impl MarkdownSummaryLine<'_> {
             return String::new();
         }
 
-        let replacer = |_: &str, s: &str| {
-            if let Some(link) = links.iter().find(|link| &*link.original_text == s) {
-                Some((link.href.clone(), link.new_text.clone()))
+        let mut replacer = |broken_link: BrokenLink<'_>| {
+            if let Some(link) =
+                links.iter().find(|link| &*link.original_text == broken_link.reference)
+            {
+                Some((CowStr::Borrowed(&link.href), CowStr::Borrowed(&link.new_text)))
```

Comment:
See my comment above:

```suggestion
                Some((link.href.as_str().into(), link.new_text.as_str().into()))
```